File: src/main/java/wraith/fabricaeexnihilo/json/ingredient/ItemIngredientJson.java
package wraith.fabricaeexnihilo.json.ingredient;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;
import com.google.gson.JsonSerializationContext;
import wraith.fabricaeexnihilo.api.crafting.ItemIngredient;
import wraith.fabricaeexnihilo.json.BaseJson;

import java.lang.reflect.Type;

public final class ItemIngredientJson extends BaseJson<ItemIngredient> {

    private ItemIngredientJson() {}

    public static final ItemIngredientJson INSTANCE = new ItemIngredientJson();

    @Override
    public ItemIngredient deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        return ItemIngredient.fromJson(json, context);
    }

    @Override
    public JsonElement serialize(ItemIngredient src, Type typeOfSrc, JsonSerializationContext context) {
        return src.toJson(context);
    }

}

File: src/main/kotlin/exnihilofabrico/json/ItemIngredientJson.kt
package exnihilofabrico.json

import com.google.gson.*
import com.google.gson.reflect.TypeToken
import exnihilofabrico.api.crafting.ItemIngredient
import java.lang.reflect.Type

object ItemIngredientJson: JsonDeserializer<ItemIngredient>, JsonSerializer<ItemIngredient> {
    val TYPE_TOKEN: Type? = object: TypeToken<ItemIngredient>(){}.type
    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext) = ItemIngredient.fromJson(json, context)

    @Throws(JsonParseException::class)
    override fun serialize(src: ItemIngredient, typeOfSrc: Type, context: JsonSerializationContext) = src.toJson(context)
}

